core/utils/compare.go

```diff
@@ -0,0 +1,9 @@
+package utils
+
+func IsZero[C comparable](val C) bool {
+	var zero C
+	return zero == val
+}
+func Equal[C comparable](val, other C) bool {
+	return val == other
+}
```

Comment:
Does this need to be defined? Isn't an inline `==` equivalent?

Reply:
good catch, will remove :)